recipes/itk/all/conanfile.py

```diff
@@ -0,0 +1,241 @@
+        cmake.definitions["Module_ITKLevelSetsv4Visualization"] = "OFF"
+        cmake.definitions["Module_ITKVtkGlue"] = "OFF"
+
+        # Disabled because Vxl vidl is not build anymore
+        cmake.definitions["Module_ITKVideoBridgeVXL"] = "OFF"
+
+        if not tools.os_info.is_windows:
+            cmake.definitions["CMAKE_POSITION_INDEPENDENT_CODE"] = "ON"
+        cmake.configure(build_folder=self.build_subfolder)
+        cmake.build()
+        cmake.install()
+
+    def cmake_fix_macos_sdk_path(self, file_path):
+        # Read in the file
+        with open(file_path, 'r') as file:
+            file_data = file.read()
+
+        if file_data:
+            # Replace the target string
+            file_data = re.sub(
+                # Match sdk path
+                r';/Applications/Xcode\.app/Contents/Developer/Platforms/MacOSX\.platform/Developer/SDKs/MacOSX\d\d\.\d\d\.sdk/usr/include',
+                '',
+                file_data,
+                re.M
+            )
+
+            # Write the file out again
+            with open(file_path, 'w') as file:
+                file.write(file_data)
+
+    def cmake_fix_path(self, file_path, package_name):
+        try:
+            tools.replace_in_file(
+                file_path,
+                self.deps_cpp_info[package_name].rootpath.replace('\\', '/'),
+                "${CONAN_" + package_name.upper() + "_ROOT}",
+                strict=False
+            )
+        except:
+            self.output.info("Ignoring {0}...".format(package_name))
+
+    def package(self):
+        for path, subdirs, names in os.walk(self.package_folder):
+            for name in names:
+                if fnmatch(name, '*.cmake'):
+                    cmake_file = os.path.join(path, name)
+                    tools.replace_in_file(
+                        cmake_file,
+                        self.package_folder.replace('\\', '/'),
+                        '${CONAN_ITK_ROOT}',
+                        strict=False
+                    )
+
+                    self.cmake_fix_path(cmake_file, "gdcm")
+
+                    if tools.os_info.is_macos:
+                        self.cmake_fix_macos_sdk_path(cmake_file)
+
+    def package_info(self):
+        self.cpp_info.builddirs = ['lib/cmake/ITK-{0}/'.format(
+            self.upstream_version)]
+        self.cpp_info.libs = tools.collect_libs(self)
+        self.cpp_info.includedirs = ['include']
+        # This hack allow to include directly, for example, a vnl
+        # file using `#include <vln/...>`
+        # Without the hack, the vln files cannot themselves include
+        # files using `#include<vnl/..>` since only <ITK-xx/vnl/...> is
+        # available. More, one has to add in CMakeLists.txt
+        # include_directories(${CONAN_INCLUDE_DIRS_ITK_SUB})
+        # to finish the use of this hack.
+        self.cpp_info.sub.includedirs = ['include/ITK-{0}'.format(
+            self.upstream_version)]
```

Comment:
@uilianries This is the other side of `include_directories(${CONAN_INCLUDE_DIRS_ITK_SUB})` in the test package. I agree this is not clever, but until now i have no other solution to let VNL components to be `#include`-ed.

Reply:
@arcadien thanks for clarifying